base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
Could use an outer constructor (and this will simplify `transpose` and `ctranspose` below).